ipalib/rpc.py

```diff
@@ -903,6 +903,12 @@ def create_connection(self, ccache=None, verbose=None, fallback=None,
         try:
             rpc_uri = self.env[self.env_rpc_uri_key]
             principal = get_principal(ccache_name=ccache)
+            stored_principal = getattr(context, 'principal', None)
+            if principal != stored_principal:
+                try:
+                    delattr(context, 'session_cookie')
+                except AttributeError:
+                    pass
```

Comment:
I *think* we should delete `context.session_cookie` also in the `except` branch of the try-except statement.